Makefile

```diff
@@ -13,34 +30,57 @@ TARGETARCH=$(shell gcc -dumpmachine)
 #
 # Added the following flags:
 # -fsanitize=address,undefined # enable ASan/UBSan
-CFLAGS = -O2 -Wall -Wformat -Wformat=2 -Wconversion -Wimplicit-fallthrough \
-  -Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 \
-	-fstack-clash-protection \
-	-fstack-protector-strong -Wl,-z,nodlopen -Wl,-z,noexecstack -Wl,-z,relro \
-	-Wl,-z,now -Wl,--as-needed -Wl,--no-copy-dt-needed-entries -Wtrampolines \
-	-Wbidi-chars=any -fPIE -pie -Werror=implicit \
-	-Werror=incompatible-pointer-types -Werror=int-conversion \
-	-fno-delete-null-pointer-checks -fno-strict-overflow -fno-strict-aliasing \
-	-fsanitize=undefined
-
-ifeq ($(TARGETARCH), x86_64-linux-gnu)
-CFLAGS += -fcf-protection=full # only supported on x86_64
+WARN_CFLAGS := -Wall -Wformat -Wformat=2 -Wconversion -Wimplicit-fallthrough \
+	-Werror=format-security -Werror=implicit -Werror=int-conversion \
+	-Werror=incompatible-pointer-types
+
+ifneq (,$(findstring gcc,$(CC_VERSION)))
+WARN_CFLAGS += -Wtrampolines -Wbidi-chars=any  # clang as for 18.1.8 doesn't support this warnings
+endif
+
+FORTIFY_CFLAGS := -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-clash-protection \
+	-fstack-protector-strong -fno-delete-null-pointer-checks \
+	-fno-strict-overflow -fno-strict-aliasing -fsanitize=undefined
+
+ifeq (yes,$(patsubst x86_64%-linux-gnu,yes,$(TARGETARCH)))
+FORTIFY_CFLAGS += -fcf-protection=full # only supporudev_rules_dirted on x86_64
 endif
-ifeq ($(TARGETARCH), aarch64-linux-gnu)
-CFLAGS += -mbranch-protection=standard # only supported on aarch64
+ifeq (yes,$(patsubst aarch64%-linux-gnu,yes,$(TARGETARCH)))
+FORTIFY_CFLAGS += -mbranch-protection=standard # only supported on aarch64
 endif
 
-ifeq (, $(shell which pkg-config))
+BIN_CFLAGS := -fPIE
+
+CFLAGS := $(WARN_CFLAGS) $(FORTIFY_CFLAGS) $(BIN_CFLAGS) $(CFLAGS)
+LDFLAGS := -Wl,-z,nodlopen -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now \
+	-Wl,--as-needed -Wl,--no-copy-dt-needed-entries -pie $(LDFLAGS)
+
+ifeq (, $(shell which $(PKG_CONFIG)))
 $(error pkg-config not installed!)
 endif
 
 all : kloak eventcap
 
 kloak : src/main.c src/keycodes.c src/keycodes.h
-	gcc -g src/main.c src/keycodes.c -o kloak -lm $(shell pkg-config --cflags --libs libevdev) $(shell pkg-config --cflags --libs libsodium) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
+	$(CC) src/main.c src/keycodes.c -o kloak -lm $(shell $(PKG_CONFIG) --cflags --libs libevdev) $(shell $(PKG_CONFIG) --cflags --libs libsodium) $(CFLAGS) $(LDFLAGS)
```

Comment:
my mistake. I actually wasn't aware it's dedicated for include paths, and thought it's alternative for `CXXFLAGS`.

fixed

Reply:
:+1:  (just fyi, since this initially confused me too, CPP in this context is "C preprocessor", not "C++", which is why it's relevant even in an application with no C++ code. I really *really* wish whoever came up with the acronyms had come up with better acronyms, but, they didn't :-/)

Reply:
ok, that's getting more embracing: I've just now connected the dots that `CPPFLAGS` are flags for `cpp` (as in executable name for C preprocessor) =))